doc/sphinx-guides/source/api/native-api.rst

```diff
@@ -87,13 +87,34 @@ Create a New Role in a Dataverse
 Creates a new role under dataverse ``id``. Needs a json file with the role description::
 
   POST http://$SERVER/api/dataverses/$id/roles?key=$apiKey
+  
```

Comment:
I find these "POST" examples that require a file to be sent across the wire to be terribly confusing unless we provide a full curl example like the `--upload-file facets.json` example above. While were in here, can we change it to be a full curl example? Should we also add a note saying that we don't really recommend creating custom roles and that you should consider opening a GitHub issue or starting a conversation on the mailing list if you are feeling the need to create a custom role?